source/src/storage/cs_IpcRamBluenet.cpp

```diff
@@ -0,0 +1,120 @@
+/*
+ * Author: Crownstone Team
+ * Copyright: Crownstone (https://crownstone.rocks)
+ * Date: Oct 24, 2022
+ * License: LGPLv3+, Apache License 2.0, and/or MIT (triple-licensed)
+ */
+
+#include <ipc/cs_IpcRamData.h>
+#include <logging/cs_Logger.h>
+#include <storage/cs_IpcRamBluenet.h>
+
+#define LogIpcRamBluenetInfo LOGi
+#define LogIpcRamBluenetDebug LOGd
+#define LogIpcRamBluenetVerbose LOGvv
+#define LogLevelIpcRamBluenetVerbose SERIAL_VERY_VERBOSE
+
+IpcRamBluenet::IpcRamBluenet() {}
+
+IpcRamBluenet& IpcRamBluenet::getInstance() {
+	// Use static variant of singleton, no dynamic memory allocation
+	static IpcRamBluenet instance;
+	return instance;
+}
+
+void IpcRamBluenet::init() {
+	// Get bluenet data from IPC ram.
+	uint8_t ipcDataSize = 0;
+
+	IpcRetCode ipcCode  = getRamData(IPC_INDEX_BLUENET_TO_BLUENET, _ipcData.raw, &ipcDataSize, sizeof(_ipcData.raw));
+	if (ipcCode != IPC_RET_SUCCESS) {
+		LogIpcRamBluenetDebug("Failed to get IPC data: ipcCode=%i", ipcCode);
```

Comment:
if this fails, don't we need to re-init the ipc ram into a valid state? Otherwise we'll never recover, right?